packages/node_modules/@node-red/registry/lib/registry.js

```diff
@@ -185,10 +185,19 @@ function loadNodeConfigs() {
 function addModule(module) {
     moduleNodes[module.name] = [];
     moduleConfigs[module.name] = module;
-    // console.log("registry.js.addModule",module.name,"user?",module.user,"usedBy",module.usedBy,"dependencies",module.dependencies)
     for (var setName in module.nodes) {
         if (module.nodes.hasOwnProperty(setName)) {
             var set = module.nodes[setName];
+            if (!set.types) {
+                const err = new Error("Node has no types")
+                err.code = "has_no_types"
+                err.details = {
+                    setName: set.name,
+                    set: { ...set }
+                }
+                console.warn(err)
```

Comment:
```suggestion
                console.warn(err)
```

Reply:
I'm not certain at this moment in time (I forget if `warn` actually outputs the extra detail). I Will need to run it again.

Reply:
ok, so I think i have a better solution for logging this.

1. add a templated i18n message
2. dont log the error at the time of finding the `set` has no `types` - simply set the `err.details` property
3. in `runtime/lib/index:162` (where `"type_already_registered"` is also logged), log an i18n ready warning message from the `err.details`

This results in the following logs...

```
29 Sep 21:25:17 - [warn] [node-red/lower-case] Set does not have any types. name: 'lower-case', module: 'node-red', file: 'C:\Users\user\repos\_node-red-issue-3861\lower-case.js'
packages/node_modules/@node-red/util/lib/log.js:94
29 Sep 21:25:31 - [warn] [node-red-node-google/google-api-config] 'google-api-config' already registered by module node-red-contrib-google-calendar
packages/node_modules/@node-red/util/lib/log.js:94
29 Sep 21:25:31 - [warn] [node-red-node-rbe/rbe] 'rbe' already registered by module node-red
packages/node_modules/@node-red/util/lib/log.js:94
29 Sep 21:25:31 - [warn] [@flowforge/nr-project-nodes/project-link] Error: Project Link nodes cannot be loaded outside of flowforge EE environment
packages/node_modules/@node-red/util/lib/log.js:94
29 Sep 21:25:57 - [warn] [lower-case/lower-case] Set does not have any types. name: 'lower-case', module: 'lower-case', file: 'C:\Users\user\repos\_node-red-issue-3861\lower-case\lower-case\lower-case.js'
packages/node_modules/@node-red/util/lib/log.js:94
29 Sep 21:26:06 - [warn] [@lowercase/lower-case2/lower-case] Set does not have any types. name: 'lower-case', module: '@lowercase/lower-case2', file: 'C:\Users\user\repos\_node-red-issue-3861\@lower-case\lower-case2\lower-case.js'
```

Resolved in [e6b3793](https://github.com/node-red/node-red/pull/3867/commits/e6b379358aa1278d261bc7bee2a1be09f9fc41f0)